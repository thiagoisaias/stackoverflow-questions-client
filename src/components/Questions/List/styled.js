import styled from "styled-components";

export const EmptyListMessage = styled.p`
  margin: 32px;
  font-size: 16px;
  color: #777;
  text-align: center;
`;

export const Wrapper = styled.div`
  margin-bottom: 32px;
`;
