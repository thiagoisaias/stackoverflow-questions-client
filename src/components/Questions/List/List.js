import React from "react";
import PropTypes from "prop-types";

import Item from "../Item/Item";
import Spinner from "../../../shared/Spinner/Spinner";

import { EmptyListMessage, Wrapper } from "./styled";
import { ErrorMessage } from "../../../shared/styled";

const List = props => {
  const { error, isLoading, questionList } = props;

  const list = questionList.map(question => {
    return <Item key={question.id} {...question} />;
  });

  if (isLoading) return <Spinner size={36} color={"#777"} />;
  if (error) return <ErrorMessage>{error}</ErrorMessage>;

  return (
    <Wrapper>
      {list.length === 0 ? (
        <EmptyListMessage>
          Nenhum resultado encontrado! Tente alterar os par√¢metros de busca.
        </EmptyListMessage>
      ) : (
        list
      )}
    </Wrapper>
  );
};

List.propTypes = {
  error: PropTypes.string,
  isLoading: PropTypes.bool.isRequired,
  questionList: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      createdAt: PropTypes.string,
      numberOfAnswers: PropTypes.number,
      user: PropTypes.shape({
        id: PropTypes.number,
        name: PropTypes.string,
        profileImage: PropTypes.string
      }),
      score: PropTypes.number,
      tags: PropTypes.arrayOf(PropTypes.string),
      title: PropTypes.string,
      url: PropTypes.string
    })
  ).isRequired
};

export default List;
