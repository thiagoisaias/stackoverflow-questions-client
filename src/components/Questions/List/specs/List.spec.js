import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import List from "../List";

describe("List", () => {
  const mockProps = {
    error: null,
    isLoading: false,
    questionList: [
      {
        id: 534,
        createdAt: "12'17",
        numberOfAnswers: 54,
        user: {
          id: 234,
          name: "Lorem",
          profileImage: "http://someurl.com"
        },
        score: 568,
        tags: ["tag1", "tag2"],
        title: "This is an example!",
        url: "http://thisisanurl.com"
      },
      {
        id: 144,
        createdAt: "13'17",
        numberOfAnswers: 59,
        user: {
          id: 134,
          name: "Foo",
          profileImage: "http://someurl.com"
        },
        score: 968,
        tags: ["tag3", "tag4"],
        title: "This is another example!",
        url: "http://thisisanurl.com"
      }
    ]
  }
  const component = shallow(<List {...mockProps} />);

  it("renders properly", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
