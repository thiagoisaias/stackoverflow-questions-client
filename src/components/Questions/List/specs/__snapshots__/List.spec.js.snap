// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders properly 1`] = `
<styled.div>
  <Item
    createdAt="12'17"
    id={534}
    key="534"
    numberOfAnswers={54}
    score={568}
    tags={
      Array [
        "tag1",
        "tag2",
      ]
    }
    title="This is an example!"
    url="http://thisisanurl.com"
    user={
      Object {
        "id": 234,
        "name": "Lorem",
        "profileImage": "http://someurl.com",
      }
    }
  />
  <Item
    createdAt="13'17"
    id={144}
    key="144"
    numberOfAnswers={59}
    score={968}
    tags={
      Array [
        "tag3",
        "tag4",
      ]
    }
    title="This is another example!"
    url="http://thisisanurl.com"
    user={
      Object {
        "id": 134,
        "name": "Foo",
        "profileImage": "http://someurl.com",
      }
    }
  />
</styled.div>
`;
